### PR TITLE
UI: redesign resource cards

### LINKED ISSUE #2242: Disabled Badge on resource cards
If a resource is disabled, show an appropriate badge/icon on the resource card to indicate it's disabled.

### LINKED ISSUE #2321: Add Workspace ID to the (i) in the workspace card in the UI
Support need to easily ask users for the workspace ID.

At present its not available, and in scenarios such as #2320, it would be very useful.

### LINKED ISSUE #2661: Add health information to footer
The footer in the UI should contain an (i) icon which will show the API version, health status, and UI version in a callout. 

- API needs to return version in the `/health` endpoint
- UI needs a version, whether semantic or git SHA - TBD

### PR DESCRIPTION
**Pull request by jjgriff93**
Resolves #2242 
Resolves #2321
Resolves #2661

## What is being addressed

I've re-designed the UI resource cards to bring them inline with the Figma designs. As part of this I've added a disabled state badge to show when a resource is disabled, added the resource Id to the info callout, re-worked the costs display to show currency symbols and also added health information to the footer callout.

image

image

image

image

image

image

image

## How is this addressed

- Changed costs display to use badge and Intel.NumberFormat for locale formatting of currencies
- Re-designed card CSS
- Re-designed status badge and added disabled state
- Added health endpoint info to footer callout
- Added resource id to resource info
- Converted resource callouts to popovers
- Updated CHANGELOG.md

### CODE REVIEW

File: ui/app/src/components/shared/CostsTag.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  resourceId: string
+}
+
+export const CostsTag: React.FunctionComponent<CostsTagProps> = (props: CostsTagProps) => {
+  const costsCtx = useContext(CostsContext);
+  const resourceCosts = costsCtx?.costs?.find((resourceCost) => {
+    return resourceCost.id === props.resourceId;
+  });
+  let costBadge = <></>;
+  if (resourceCosts) {
```
Comment (damoodamoo) [inline ui/app/src/components/shared/CostsTag.tsx:15]: you might have a bug here when there are no costs available? I'm seeing a page failure on Shared-Services, with the console log:
image

File: ui/app/src/components/shared/ResourceCard.tsx [hunk 2]
```diff
@@ -40,122 +40,116 @@ export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props:
...
-  switch(props.resource.resourceType) {
-    case ResourceType.Workspace:
-    case ResourceType.WorkspaceService:
-    case ResourceType.UserResource:
-      resourceUrl = props.resource.resourcePath;
-      break;
-    case ResourceType.SharedService: // shared services are accessed from the root and the workspace, have to handle the URL differently
-      resourceUrl = workspaceCtx.workspace ? props.resource.id : props.resource.resourcePath;
-      break;
+  const goToResource = useCallback(() => {
```
Comment (damoodamoo) [inline ui/app/src/components/shared/ResourceCard.tsx:43]: `nit` the `use` hooks want to go above any imperative code - think should move to line 34

File: ui/app/src/components/shared/ResourceCard.tsx [hunk 3]
```diff
@@ -40,122 +40,116 @@ export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props:
...
+    props.resource.azureStatus?.powerState &&
+    successStates.includes(resourceStatus) &&
+    props.resource.isEnabled
+  ) {
+    headerBadge = <PowerStateBadge state={props.resource.azureStatus.powerState} />
+  } else {
+    headerBadge = <StatusBadge resource={props.resource} status={resourceStatus} />
   }
 
+  const authNotProvisioned = props.resource.resourceType === ResourceType.Workspace && !props.resource.properties.scope_id;
```
Comment (damoodamoo) [inline ui/app/src/components/shared/ResourceCard.tsx:77]: i'm not seeing this tooltip on non-auth provisioned cards? also - can we prevent the animation / cursor changing to a pointer if the auth hasn't been provisioned? currently it's giving the signals that it's clickable, but nothing happens when I click

File: ui/app/src/components/shared/CostsTag.tsx (hunk 1 above)
Comment (jjgriff93) [reply ui/app/src/components/shared/CostsTag.tsx:15]: Ah good catch, I shall look at that now thanks!

Comment (jjgriff93) [reply ui/app/src/components/shared/CostsTag.tsx]: Think I've fixed it but can't test as I have costs available - could you pull and check?

File: ui/app/src/components/shared/ResourceCard.tsx (hunk 3 above)
Comment (jjgriff93) [reply ui/app/src/components/shared/ResourceCard.tsx:77]: Have you got the latest? Added a fix for that not long ago

Comment (jjgriff93) [reply ui/app/src/components/shared/ResourceCard.tsx:77]: You should see this: 
image

File: ui/app/src/components/shared/ResourceCard.tsx (hunk 2 above)
Comment (jjgriff93) [reply ui/app/src/components/shared/ResourceCard.tsx:35]: Fixed

File: ui/app/src/components/shared/ResourceCard.tsx (hunk 3 above)
Comment (damoodamoo) [reply ui/app/src/components/shared/ResourceCard.tsx:77]: yep, see it now, how odd as I pulled this morning. Like it!

File: ui/app/src/components/shared/CostsTag.tsx (hunk 1 above)
Comment (damoodamoo) [reply ui/app/src/components/shared/CostsTag.tsx]: yup, looks good!

**Review by damoodamoo — APPROVED** [review]
love it!

**Pull request merged**